packages/back-end/src/util/mongo.util.ts

```diff
@@ -147,7 +146,12 @@ export function removeMongooseFields<T>(
   if (doc.toJSON) {
     doc = doc.toJSON({ flattenMaps: true });
   }
-  return omit(doc, ["_id", "__v"]) as T;
+
+  // Copy the object and delete mongoose fields rather than using lodash.omit for perf reasons since this is called a lot
+  const result = { ...doc } as MongooseDocument & T;
```

Comment:
`MongooseDocument & T` isn't the right type here.  After the `toJSON()` call above, it's no longer a MonoogseDocument.

I think `T & {_id?: string; __v?: string}` is more accurate.